idaes/models/unit_models/heat_exchanger_1D.py

```diff
@@ -776,6 +805,7 @@ def initialize_build(
         cold_side_units = (
             self.cold_side.config.property_package.get_metadata().get_derived_units
         )
+        # TODO What if there is more than one time point? What if t0 != 0?
```

Comment:
This should at least have `t0 = self.flowsheet().time.first()`.

As for multiple time points, that is on the user to determine how best to proceed. Initialization generally focuses on steady-state. Note that we would probably need to update the `Initializer` as well.